### PR TITLE
JsonParser needs to have a way to get current event

### PR DESCRIPTION
**Pull request by jbescos**
Relates to https://github.com/eclipse-ee4j/jsonp/issues/282

### CODE REVIEW

**Comment by lukasj** [comment]
@Verdent is this what you are looking for?

**Comment by Verdent** [comment]
@jbescos @lukasj It is :-) Thank you! Just wondering, why is it default?

**Comment by jbescos** [comment]
> @jbescos @lukasj It is :-) Thank you! Just wondering, why is it default?

In this way, the implementations of jsonp can compile without implementing this method immediately.

I am not sure if this is something we need to take care, but I saw there are already other default methods in the interface, so I was following that convention.

**Comment by lukasj** [comment]
using _default_ is the only way to not break backward compatibility of the interface; w/o that every vendor would be required to implement it ...

**Comment by Verdent** [comment]
Sure, I was just wondering :-)

File: api/src/main/java/jakarta/json/stream/JsonParser.java [hunk 1]
```diff
@@ -475,6 +475,17 @@ default public void skipObject() {
...
+    /**
+     * Gets the event in course.
+     *
+     * @return the current event.
+     
+     * @since 2.1
+     */
+    default public Event currentEvent() {
+        throw new UnsupportedOperationException();
+    }
```
Comment (lukasj) [inline api/src/main/java/jakarta/json/stream/JsonParser.java:487]: I'd move this next to other event related methods (likely after the `next()`) and improved javadoc to look more consistent with the rest. Ie return could be updated to `@return the event for the current parsing state`

**Review by lukasj — APPROVED** [review]
Thanks!

**Pull request merged**